core/dbt/version.py

```diff
@@ -27,81 +51,175 @@ def get_latest_version(version_url: str = PYPI_VERSION_URL):
-        plugin_update_msg = ""
-        if installed == plugin_version or (
-            latest_plugin_version and plugin_version == latest_plugin_version
-        ):
-            compatibility_msg = green("Up to date!")
-        else:
-            if latest_plugin_version:
-                if installed.major == plugin_version.major:
-                    compatibility_msg = yellow("Update available!")
-                else:
-                    compatibility_msg = red("Out of date!")
-                plugin_update_msg = (
-                    "  Your version of dbt-{} is out of date! "
-                    "You can find instructions for upgrading here:\n"
-                    "  https://docs.getdbt.com/dbt-cli/install/overview\n\n"
-                ).format(plugin_name)
-            else:
-                compatibility_msg = yellow("No PYPI version available")
-
-        plugin_version_msg += ("  - {}: {} - {}\n" "{}").format(
-            plugin_name, version, compatibility_msg, plugin_update_msg
-        )
+    for name, version, update_msg in _pad_lines(lines, seperator=":"):
+        line_msg = f"  - {name} {version}"
+        if update_msg != "":
+            line_msg += f" - {update_msg}"
+        msg_lines.append(line_msg)
 
-    if latest is None:
-        return (
-            "{}The latest version of dbt could not be determined!\n"
-            "Make sure that the following URL is accessible:\n{}\n\n{}".format(
-                version_msg, PYPI_VERSION_URL, plugin_version_msg
-            )
-        )
+    return msg + "\n".join(msg_lines)
+
+
+def _get_plugins_msg(installed: Type[dbt.semver.VersionSpecifier]) -> str:
+    msg = "Plugins:\n"
+    msg_lines = []
 
-    if installed == latest:
-        return f"{version_msg}{green('Up to date!')}\n\n{plugin_version_msg}"
+    plugins = []
+    plugin_msgs = []
+    for name, version_s in _get_dbt_plugins_info():
+        compatability_msg, update_msg = _get_plugin_msg_info(name, version_s, installed)
+        plugins.append([name, version_s, compatability_msg])
+        plugin_msgs.append(update_msg)
 
-    elif installed > latest:
-        return "{}Your version of dbt is ahead of the latest " "release!\n\n{}".format(
-            version_msg, plugin_version_msg
+    for i, plugin in enumerate(_pad_lines(plugins, seperator=":")):
+        msg_lines.append(_format_single_plugin(plugin, plugin_msgs[i]))
+    return msg + "\n".join(msg_lines)
+
+
+def _get_plugin_msg_info(
+    name: str, version_s: str, core: Type[dbt.semver.VersionSpecifier]
+) -> Tuple[str, str]:
+    plugin = dbt.semver.VersionSpecifier.from_version_string(version_s)
+    latest_plugin = get_latest_version(version_url=get_package_pypi_url(name))
+
+    update_msg = ""
+
+    if plugin.major != core.major and plugin.minor != core.minor:
+        compatibility_msg = red("Not compatible!")
+        update_msg = (
+            f"  Your version of dbt-{name} is not compatible with core!\n"
```

Comment:
[Not blocking, cosmetic only] This feels a little clearer to me:
```suggestion
            f"  The installed versions of dbt-core and dbt-{name} are not compatible.\n"
```

Reply:
Similar to my question below, but how often do users call `dbt` `dbt-core`?

Reply:
Good question!
- People call this `dbt-core` to distinguish it from the dbt adapter plugin (`dbt-postgres`, `dbt-redshift`, ...)
- For trademark purposes, there exists an important distinction between "dbt" (Core + Cloud) and "dbt Core" (hence why we renamed this repo)

Reply:
That makes sense!